src/Particle/Particle.cpp

```diff
@@ -0,0 +1,356 @@
+/* Copyright 2022 The AMReX Community
+ *
+ * Authors: Ryan Sandberg
+ * License: BSD-3-Clause-LBNL
+ */
+#include <pybind11/pybind11.h>
+#include <pybind11/operators.h>
+#include <pybind11/stl.h>
+
+#include <AMReX_Config.H>
+#include <AMReX_BoxArray.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_RealVect.H>
+#include <AMReX_Particle.H>
+
+#include <array>
+#include <stdexcept>
+#include <string>
+#include <sstream>
+#include <stdexcept>
+#include <utility>
+#include <cmath>
+#include <regex>
+
+
+
+namespace py = pybind11;
+using namespace amrex;
+using pReal = amrex_particle_real;
+
+struct PIdx
+{
+    enum RealValues { // Particle Attributes for sample particle struct-of-arrays
+          w = 0,
+          vx, vy, vz,
+          Ex, Ey, Ez,
+          nRealAttribs
+    };
+
+    enum IntValues {
+        nIntAttribs
+    };
+
+};
+
+namespace
+{
+    /** Build a std::array from a fixed-size C array at compile-time */
+    template<typename T, std::size_t... I>
+    constexpr auto
+    build_array (T a[], std::index_sequence<I...> s)
+    {
+        return std::array<T, s.size()>{a[I]...};
+    }
+}
+
+template <int T_NReal, int T_NInt=0>
+void make_Particle(py::module &m)
+{
+    using ParticleType = Particle<T_NReal, T_NInt>;
+    auto const particle_name = std::string("Particle_").append(std::to_string(T_NReal) + "_" + std::to_string(T_NInt));
+    py::class_<ParticleType> (m, particle_name.c_str())
+        .def(py::init<>())
+        .def(py::init([](AMREX_D_DECL(ParticleReal x, ParticleReal y, ParticleReal z)) { 
+                    std::unique_ptr<ParticleType> part(new ParticleType());
+                    // AMREX_D_DECL(part->m_pos[0] = x;, part->m_pos[1] = y;, part->m_pos[2] = z;)
+                    part->m_pos[0] = x;
+    #if (AMREX_SPACEDIM >= 2)
+                    part->m_pos[1] = y;
+    #endif
+    #if (AMREX_SPACEDIM >= 3)
+                    part->m_pos[2] = z;
+    #endif
+                    return part;
+                }
+            )
+        )
+        .def(py::init([](AMREX_D_DECL(ParticleReal x, ParticleReal y, ParticleReal z), py::args& args) { 
+                    std::unique_ptr<ParticleType> part(new ParticleType());
```

Comment:
somehow weirdly deep aligned block of code compared to previous lines?

Reply:
trying to help myself out when writing the lambda